ui/packages/consul-nspaces/app/components/consul/nspace/selector/README.mdx

```diff
@@ -0,0 +1,40 @@
+# Consul::Nspace::Selector
+
+A conditional, autoloading, menu component specifically for making it easy to select namespaces.
+
+Please note: 
+
+- Currently at least, you must add this inside of a `<ul>` element.
+- For the moment, make sure you have enabled nspaces using developer debug
+  cookies.
+
+```hbs preview-template
+<ul>
+  <Consul::Nspace::Selector
+    @dc={{hash
+      Name='dc-1'
+    }}
+    @nspace='default'
+    @partition='default'
+    @nspaces={{or this.nspaces (array)}}
+    @onchange={{action (mut this.nspaces) value="data"}}
+  />
+</ul>
```

Comment:
Yeah I think you got it there from what you explained, ideally it would know whether it was in a ul or not and autoadd a li if so. Another approach would be to `display: none` empty `<li>`s - but thats a bit yuk - but we may or may not change to do that at somepoint, plus you could get caught up in whitespace issues where the `li` is empty but full of handlebars related whitespace (where `~` sometimes works and sometimes doesn't)

I think the suggestion you added below captures it perfectly so I'll add that in here 👍